docs/components/select.md

```diff
@@ -1,1013 +1,741 @@
-By default, the loading indicator is a spinner icon, and you can implement your own indicator using the `loading` slot.
-See [autosuggest](#autosuggest) for an example.
+Use fixed positioning of the popover to avoid content being clipped by parental boundaries. See [KPopover](/components/popover#positionfixed) docs for more information.
 
-### reuseItemTemplate
+### enableItemCreation
 
-Use this prop to customize selected item element appearance by reusing content passed through [item-template](#item-template) slot. **Note:** this prop only applies when `appearance` prop is `select` (use [selected-item-template](#selected-item-template) slot in other cases).
+When used in conjunction with `enableFiltering` set to `true`, KSelect will suggest adding a new value when filtering produces no results.
 
 <ClientOnly>
-  <KSelect reuse-item-template appearance="select" :items="deepClone(defaultItems)">
-    <template #item-template="{ item }">
-      <div class="item-template-container">
-        <span v-if="item.value === 'cats'">🐈</span>
-        <span v-if="item.value === 'dogs'">🐕</span>
-        <span v-if="item.value === 'bunnies'">🐇</span>
-        <div class="select-item-label">{{ item.label }}</div>
-      </div>
-    </template>
-  </KSelect>
+  <KSelect enable-item-creation enable-filtering placeholder="Try searching for 'service d'" :items="selectItemsUnselected" />
 </ClientOnly>
 
 ```html
-<KSelect reuse-item-template appearance="select" :items="items">
-  <template #item-template="{ item }">
-    <div>
-      <span v-if="item.value === 'cats'">🐈</span>
-      <span v-if="item.value === 'dogs'">🐕</span>
-      <span v-if="item.value === 'bunnies'">🐇</span>
-      <div class="select-item-label">{{ item.label }}</div>
-    </div>
-  </template>
-</KSelect>
+<KSelect enable-item-creation enable-filtering placeholder="Try searching for 'service d'" :items="selectItems" />
 ```
 
-### enableItemCreation
-
-`KSelect` offers users the ability to add custom item to the list by typing the item they want to and then clicking the `... (Add new value)` item at the bottom of the list, which will also automatically select it.
-
-Newly created item will have a `label` consisting of the user input and a randomly generated id for the `value` to ensure uniqueness. It will also have an attribute `custom` set to `true`. This action triggers an `item:added` event containing the added item data.
+### loading
 
-Deselecting the item will completely remove it from the list and underlying data, and trigger a `item:removed` event containing the removed item's data.
+Pass `true` to display loader instead of items in the dropdown. KSelect's `item` prop is reactive to changes by design, so that should you need to perform async item fetching/filtering you can execute that logic on host app's side and pass items back to KSelect. The example below utilizes the [`@query-change` event](#query-change) to simulate async items fetching behind the scenes.
```

Comment:
```suggestion
Pass `true` to display loader instead of items in the dropdown. KSelect's `item` prop is reactive to changes by design, so that should you need to perform async item fetching/filtering you can execute that logic within the host app and pass items back to KSelect. 

The example below utilizes the [`@query-change` event](#query-change) to simulate async item fetching behind the scenes.
```